fili-core/src/main/java/com/yahoo/bard/webservice/web/SlicesApiRequest.java

```diff
@@ -165,7 +164,7 @@ protected SlicesApiRequest() {
             throw new BadApiRequestException(msg);
         }
 
-        Map<Column, List<Interval>> columnCache = table.getAvailability().getAvailableIntervals();
+        Map<Column, Set<Interval>> columnCache = table.getAllAvailableIntervals();
```

Comment:
It's amazingly hard to write tests involving non-trivial JSON objects where some parts are order-agnostic, while others are order-sensitive. If you can think of a simple solution, please do share it because it would save so much pain.

Without that, however, the more straight-forward solution is to be generally stricter around enforcing ordering outbound. Given how much pain these issues have caused, and the difficulty in debugging them (since it's different behavior on different hosts), I'm not willing to risk dropping the ordering if there isn't a compelling reason, especially for already existing interfaces.